ICAFIX/scripts/run_icasso.m

```diff
@@ -0,0 +1,253 @@
+%   nICA    : Number of ICA repetitions per icasso repetition, @ delimited string, (default = '100')
+%   nClust  : Number of icasso repetitions (default = '1')
+%   maxIter : Maximum number of iterations per ica fit (default = '1000')
+%
+% example inputs:
+% Dim = '41';
+% concatfmri = 'tfMRI_EMOTION_RL_LR';
+% concatfmrihp = 'tfMRI_EMOTION_RL_LR_hp0';
+% ConcatFolder = '/mnt/myelin/burke/HCPpipelines/dev_study/100307/MNINonLinear/Results/tfMRI_EMOTION_RL_LR_ICASSO';
+% vis = 'basic';
+% nICA = '2@2';
+% nClust = '2';
+% maxIter = '1000';
+% 
+% Dim = str2double(Dim);
+% nICA = cellfun(@str2double,regexp(nICA,'@','split'));
+% nClust = str2double(nClust);
+% maxIter = str2double(maxIter);
+
+% Created 2024-06-05
+% Burke Rosen
+
+% ToDo:
+% () Currently the vnts file and brainMaskFile paths and names are inferred from
+%    ConcatFolder, concatfmrihp, and concatfmri. Maybe they should be their own 
+%    arguments for flexibility.
+
+%% parse parameters
+if nargin < 4 || any(cellfun(@isempty,{Dim,concatfmri,concatfmrihp,ConcatFolder}))
+  error('Dim, concatfmri, concatfmrihp, and ConcatFolder are required!')
+end
+Dim = str2double(Dim);
+if nargin < 5 || isempty(vis) 
+  vis = 'basic';
+end
+if ~ismember(vis,{'basic','off'})
+  warning('vis must be ''basic'' or ''off'', reverting to ''basic''')
+  vis = 'basic';
+end
+if nargin < 6 || isempty(nICA) 
+  nICA = 100;
+else
+  nICA = cellfun(@str2double,regexp(nICA,'@','split'));
+end
+if nargin < 7 || isempty(nClust) 
+  nClust = 1;
+else
+  nClust = str2double(nClust);
+end
+if numel(nICA) ~= nClust
+  error('nICA must be a @ delimited list with a number of elements equal to nClust')
+end
+if nargin < 8 || isempty(maxIter) 
+  maxIter = 1000;
+else
+  maxIter = str2double(maxIter);
+end
+
+%% check IO dependencies
+% use imaging processing toolbox utilities, or if those are not available use FSL utilities 
+function out = out2(fun);[~,out] = fun();end
+function out = out3(fun);[~,~,out] = fun();end
+if isempty(which('niftiread'))
+  infoNIFTI = @(fName) struct('ImageSize',out2(@() read_avw(fName))','PixelDimensions',out3(@() read_avw(fName))');
+  readNIFTI = @(fName) read_avw(fName);
+  writeNIFTI = @(img,fName,hdr) save_avw(img,fName,'f',hdr.PixelDimensions);
+  if isempty(which('read_avw'))
+    error('neither niftiread nor read_avw on matlab path!')
+  end
+else
+  readNIFTI = @(fName) niftiread(fName);
+  infoNIFTI = @(fName) niftiinfo(fName);
+  writeNIFTI = @(img,fName,hdr) niftiwrite(img,fName,hdr);
+end
+
+%% parse paths
+% inputs
+vntsFile = sprintf('%s/%s_vnts.nii.gz',ConcatFolder,concatfmrihp);
+brainMaskFile = sprintf('%s/%s_brain_mask.nii.gz',ConcatFolder,concatfmri);
+if ~exist(vntsFile,'file');error('%s doesn''t exist!',vntsFile);end
+if ~exist(brainMaskFile,'file');error('%s doesn''t exist!',brainMaskFile);end
+
+% outputs
+outDir = sprintf('%s/%s.ica/filtered_func_data.ica',ConcatFolder,concatfmrihp);
+[~,~] = unix(['mkdir -p ' outDir]);
```

Comment:
https://www.mathworks.com/help/matlab/ref/mkdir.html

Please don't use shell calls, and particularly not the overly-specific `unix()`, to do filesystem stuff that is built into the language.  You should be able to use the status-outputting function form, `if ~mkdir(outDir); error('unable to make output folder'); end` to avoid the "already exists" warning and provide an informative message.

Reply:
I'll swapped unix for system moving forward. That's a neat mkdir trick, revised.